packages/vscode-dependencies-validation/src/extension.ts

```diff
@@ -1,27 +1,96 @@
-import { Uri, ExtensionContext, workspace, window } from "vscode";
 import {
-  NPMDependencyIssue,
+  Uri,
+  ExtensionContext,
+  workspace,
+  window,
+  languages,
+  DiagnosticCollection,
+  OutputChannel,
+  commands,
+  CodeActionKind,
+} from "vscode";
+import {
+  NpmLsResult,
   findDependencyIssues,
 } from "@sap-devx/npm-dependencies-validation";
+import { NPMIssuesActionProvider } from "./npmIssuesActionProvider";
+import { fixAllDepIssuesCommand } from "./commands";
+import { FIX_ALL_ISSUES_COMMAND } from "./constants";
+import { refreshDiagnostics } from "./diagnostics";
+
+const PACKAGE_JSON = "package.json";
+const PACKAGE_JSON_PATTERN = `**​/${PACKAGE_JSON}`;
+const extName = "vscode-dependency-validation";
 
 export function activate(context: ExtensionContext) {
+  const outputChannel = window.createOutputChannel(extName);
+  const diagnosticCollection = createDiagnosticCollection(context, extName);
+
   void findIssues();
 
   void addFileWatcher();
+
+  registerCodeActionsProvider(context);
+
+  subscribeToDocumentChanges(context, diagnosticCollection);
+
+  registerCommands(context, outputChannel, diagnosticCollection);
+}
+
+function registerCodeActionsProvider(context: ExtensionContext): void {
+  context.subscriptions.push(
+    languages.registerCodeActionsProvider(
+      { language: "json", scheme: "file", pattern: "**/package.json" },
```

Comment:
did not we build this pattern already above in a cost?

Reply:
I have redefined the pattern but here we still need to use this one